src/main/java/com/microsoft/sqlserver/jdbc/DDC.java

```diff
@@ -1060,6 +1064,136 @@ static final Object convertTemporalToObject(JDBCType jdbcType, SSType ssType, Ca
         }
     }
 
+    private static Object convertTemporalToObject(JDBCType jdbcType, SSType ssType, int daysSinceBaseDate,
+            long ticksSinceMidnight, int fractionalSecondsScale) {
+        int subSecondNanos;
```

Comment:
Is subSecondNanos required anymore? I tried removing a few setNanos() calls and didn't seem to break anything.

Reply:
That variable is the millisecond value after DD:MM:SS. JUnit tests should fail without this variable.

Reply:
Wouldn't plusNanos() already take care of that?

Reply:
plusNanos() is a way to offload the calculation of converting the amount of nanoseconds from midnight to LocalDateTime, whereas subSecondNanos is the actual variable that holds the nanosecond values after DD:MM:SS. plusNanos already gives the LocalDateTime values with the correct nanosecond values, but depending on the target datatype, the nanosecond value inside LocalDateTime might be lost during conversion from LocalDateTime (for example, cases where the target datatype is java.sql.Timestamp), so we need this variable to hold onto the nanosecond value.